samples/sensor/thermometer/sample.yaml

```diff
@@ -15,6 +15,8 @@ tests:
       - platform:mimxrt1180_evk/mimxrt1189/cm7:SHIELD=p3t1755dp_ard_i2c
       - platform:mimxrt1180_evk/mimxrt1189/cm33:SHIELD=p3t1755dp_ard_i3c
       - platform:mimxrt1180_evk/mimxrt1189/cm7:SHIELD=p3t1755dp_ard_i3c
+      - platform:frdm_mcxn236:SHIELD=p3t1755dp_ard_i2c
+      - platform:frdm_mcxn236:SHIELD=p3t1755dp_ard_i3c
```

Comment:
sorry, I miss check luient's pr for this part, you can't assign the same platform with i2c, and i3c at the same time, this will cause problem, you need add another test node like below
```
common:
    filter: dt_alias_exists("ambient-temp0")
    harness: console
    harness_config:
      fixture: sensor_ambient_temp
      type: multi_line
      regex:
        - "Thermometer Example (.*)"
        - "Temperature device is 0x[0-9|a-z]+, name is [a-z|0-9]+@[a-z|0-9]+"
        - "Temperature is [0-9|.]+°C"
tests:
  sample.sensor.thermometer:
    tags: sensors
    fixture:
      - fixture_i2c_sensor 
    integration_platforms:
      - nrf52840dk/nrf52840   # mcp9700a
      - frdm_k22f             # tcn75a
      - robokit1              # ntc_thermistor
      - adi_eval_adin1110ebz  # adt7420
    extra_args:
      - platform:mimxrt1180_evk/mimxrt1189/cm33:SHIELD=p3t1755dp_ard_i2c
      - platform:mimxrt1180_evk/mimxrt1189/cm7:SHIELD=p3t1755dp_ard_i2c
      - platform:frdm_mcxn236:SHIELD=p3t1755dp_ard_i2c
  sample.sensor.thermometer_i3c:
    tags: sensors
    integration_platforms:
      - frdm_mcxn947/mcxn947/cpu0 # p3t1755
    depends_on:
    - i3c
    - sensor
    filter: dt_alias_exists("ambient-temp0")
    fixture: fixture_i3c_sensor 
    extra_args:
      - platform:mimxrt1180_evk/mimxrt1189/cm33:SHIELD=p3t1755dp_ard_i3c
      - platform:mimxrt1180_evk/mimxrt1189/cm7:SHIELD=p3t1755dp_ard_i3c
      - platform:frdm_mcxn236:SHIELD=p3t1755dp_ard_i3c
```

Reply:
also please add i3c support to `boards/nxp/frdm_mcxn236/frdm_mcxn236.yaml`